src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -74,8 +74,9 @@ public bool StartAssignment(HostAssignmentContext assignmentContext)
                     assignmentContext.ApplyAppSettings();
                 }
 
-                // Restart runtime.
-                _scriptHostManager.RestartHost();
+                // set flags which will trigger specialization
+                _settingsManager.SetSetting(EnvironmentSettingNames.AzureWebsitePlaceholderMode, "0");
```

Comment:
This coupled with the Startup class change I made is how placeholder/specialization is controlled.